machineconfiguration/v1/types.go

```diff
@@ -759,3 +760,123 @@ type ContainerRuntimeConfigList struct {
+	Config MachineConfigStateConfig `json:"configuration"`
+	// mostRecentState is the most recent state reporting for each object in a particular machineState
+	// +listType=map
+	// +listMapKey=name
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	MostRecentState []ProgressionCondition `json:"mostRecentState" patchStrategy:"merge" patchMergeKey:"name"`
+	// progressionHistory contains a list of events that have happened on all objects in this machinestate
+	ProgressionHistory []ProgressionHistory `json:"progressionHistory"`
+	// mostRecentError is populated if the State reports an error.
+	MostRecentError string `json:"mostRecentError"`
+	// health reports the overall status of this MachineConfigState Given its Progress
+	Health MachineConfigStateHealthEnumeration `json:"health"`
+}
+
+type MachineConfigStateHealthEnumeration string
+
+const (
+	// healthy describes a machineState that is functioning properly
+	Healthy MachineConfigStateHealthEnumeration = "Healthy"
+	// unknown describes a machieState who's health is unknown
+	Unknown MachineConfigStateHealthEnumeration = "Unknown"
+	// unhealthy describes a machinestate that is not functioning properly
+	UnHealthy MachineConfigStateHealthEnumeration = "Unhealthy"
+)
+
+// MachineConfigStateConfig describes the configuration of a MachineConfigState
+type MachineConfigStateConfig struct {
+	corev1.ObjectReference `json:",inline"`
+}
+
+// StateProgress is each possible state for each possible MachineConfigStateType
+// UpgradeProgression Kind will only use the "MachinConfigPoolUpdate..." types for example
+type StateProgress string
+
+const (
+	// MachineConfigPoolUpdatePreparing describes a machine that is preparing in the daemon to trigger an update
+	MachineConfigPoolUpdatePreparing StateProgress = "UpdatePreparing"
+	// MachineConfigPoolUpdateInProgress describes a machine that is in progress of updating
+	MachineConfigPoolUpdateInProgress StateProgress = "UpdateInProgress"
+	// MachineConfigPoolUpdatePostAction describes a machine that is executing its post update action
+	MachineConfigPoolUpdatePostAction StateProgress = "UpdatePostAction"
+	// MachineConfigPoolUpdateCompleting describes a machine that is in the process of resuming normal processes
+	MachineConfigPoolUpdateCompleting StateProgress = "UpdateCompleting"
+	// MachineConfigPoolUpdateComplete describes a machine that has a matching desired and current config after executing an update
+	MachineConfigPoolUpdateComplete StateProgress = "Updated"
+	// MachineConfigPoolUpdateResuming describes a machine that is in the process of resuming normal processes
+	MachineConfigPoolResuming StateProgress = "Resuming"
+	// MachineConfigPoolReady describes a machine that has a matching desired and current config
+	MachineConfigPoolReady StateProgress = "Ready"
+	// MachineConfigStateErrored describes when a machine had run into an issue
+	MachineConfigStateErrored StateProgress = "Errored"
+)
```

Comment:
Are these all still relevant for the MCD specific MSC API?

Reply:
Yep, I have a working version and all of these get used.